internal/app/subsystems/api/grpc/api/schedule.proto

```diff
@@ -0,0 +1,81 @@
+syntax = "proto3";
+
+package schedule;
+
+option go_package = "github.com/resonatehq/resonate/internal/app/subsystems/api/grpc/api";
+
+message PromiseValue {
+    bytes data = 1; 
+    map<string, string> headers = 2;   
+}
+
+message Schedule {
+    string id = 1;
+    string description = 2; 
+    string cron = 3;
+    map<string, string> tags = 4;
+    string promiseId = 5;
+    int64 promiseTimeout = 6;
+    PromiseValue promiseParam = 7;
+    map<string, string> promiseTags = 8;
+    int64 lastRunTime = 9;
+    int64 nextRunTime = 10; 
+    string idempotencyKey = 11;
+    int64 createdOn = 12;    
+}
+
+// CREATE 
+
+message CreateScheduleRequest {
+    Schedule schedule = 1; 
```

Comment:
I think we should manually list the fields that can be specified here since it is only a subset

Reply:
this is not the only place we use schedule type. that would just give us a lot of boilerplate (copy paste) every the type schedule is used. Plus, for protos it's better to do this since we can change the schema without breaking as much as if it was inline.

Reply:
This is not consistent with promises, see [CreatePromiseRequest](https://github.com/resonatehq/resonate/blob/main/internal/app/subsystems/api/grpc/api/promise.proto#L63-L70) for example, we have a subset of the Promise object.

Reply:
that's openapi. look at current promise.proto

Reply:
oh i see what ur saying now. was misundertanding before. let's chat about this. after if we want the experience of grpc to be consistent with http experieence we should prob update. we do similar with status. let's chat.